lr_face/data.py

```diff
@@ -732,14 +759,57 @@ def make_pairs(data: Union[Dataset, List[FaceImage]],
             # negative pairs that contain at least one of the images from each
             # positive pair and a randomly chosen other image with a different
             # identity.
-            for first, second in res.copy():  # Copy, because we modify `res`.
+            for first, second in result.copy():  # Copy, because we modify `result`.
                 identity = first.identity
                 negative_id = random.choice(tuple(identities - {identity}))
                 negative = random.choice(images_by_identity[negative_id])
-                res.append(FacePair(random.choice([first, second]), negative))
+                result.append(FacePair(random.choice([first, second]), negative))
 
         if n:
-            res = random.sample(res, min(len(res), n))
+            result = random.sample(result, min(len(result), n))
+    return result
+
+
+def make_pairs_from_two_lists(
+        data_first: List[FaceImage],
+        data_second: List[FaceImage],
+        n: Optional[int] = None) -> List[FacePair]:
+    """
+    Takes two list of `FaceImage` instances and pairs them up, each pair
+    having one image from the first, one from the seconds
+
+    Returns:
+        A list of `FacePair` instances.
+    """
+    images_first_by_identity = defaultdict(list)
+    for x in data_first:
+        images_first_by_identity[x.identity].append(x)
+
+    images_second_by_identity = defaultdict(list)
+    for x in data_second:
+        images_second_by_identity[x.identity].append(x)
+
+    res = []
+    for identity in set(images_first_by_identity.keys()).intersection(
```

Comment:
ik was dit nog eens aan het uittesten, en het lijkt erop dat dezelfde images in data_first en data_second kunnen zitten, waardoor je hier een image met zichzelf gaat vergelijken. dat lijkt me niet handig

Reply:
regel 797 eerst, if image_a != image_b: